src/YourEasyRent/DataBase/UserStateManagerRepository.cs

```diff
@@ -0,0 +1,56 @@
+﻿using MongoDB.Driver;
+using Telegram.Bot.Types;
+using YourEasyRent.DataBase.Interfaces;
+using YourEasyRent.Entities;
+using YourEasyRent.UserState;
+
+namespace YourEasyRent.DataBase
+{
+    public class UserStateManagerRepository : IUserStateManagerRepository
+    {
+        private readonly IMongoCollection<UserSearchStateDTO> _collectionOfUserSearchState;
+        public UserStateManagerRepository(DataBaseConfig dateBaseConfig, IMongoClient mongoClient)
+        {
+            var dataBase = mongoClient.GetDatabase(dateBaseConfig.DataBaseName);
+            _collectionOfUserSearchState = dataBase.GetCollection<UserSearchStateDTO>(dateBaseConfig.CollectionName);
+        }
+
+        public async Task<UserSearchState> GetForUser(string userId)
+        {
+
+            
+            var filter = Builders<UserSearchStateDTO>.Filter.Eq(u => u.UserId.ToString(), userId);// ошибка MongoDB.Driver.Linq.ExpressionNotSupportedException: 'Expression not supported: u.UserId.ToString().'
+
+            var dto = await _collectionOfUserSearchState.Find(filter).FirstOrDefaultAsync();
+
+            var state = new UserSearchState(dto);
+
+            return state;
+        }
+
+        public async Task CreateAsync(UserSearchState userSearchState)//  мы должны выполнить этот метод когда все поля будут заполены или 
+            // достаточно и одного поля??
+        {
+            var dto = userSearchState.TOMongoRepresintation();
+            await _collectionOfUserSearchState.InsertOneAsync(dto);
+        }
+
+        public async Task UpdateAsync(UserSearchState userSearchState)
+        {
+            var filter = Builders<UserSearchStateDTO>.Filter.Eq(u => u.UserId, userSearchState.UserId);
+            var dto = userSearchState.TOMongoRepresintation();
+            await _collectionOfUserSearchState.ReplaceOneAsync(filter, dto);
+        }
+
+        public async Task<MenuStatus> GetCurrentStateForUser(string userId)
```

Comment:
You don't need this method - jsut use GetForUser(string userId)

Reply:
You told me to create this method. I thought we should use this method to get the current status and, based on it, take a “step back” or end the chat maybe

Reply:
You can do it this way, or you can just resue GetForUser

```
var state = await GetForUser(string userId);
var currentStatus = state.Status;
```